plutus-playground-server/app/PSGenerator.hs

```diff
@@ -294,22 +325,82 @@ multilineString :: Text -> Text -> Text
 multilineString name value =
     "\n\n" <> name <> " :: String\n" <> name <> " = \"\"\"" <> value <> "\"\"\""
 
+jsonExport :: ToJSON a => Text -> a -> Text
+jsonExport name value =
+    multilineString name (T.decodeUtf8 . BSL.toStrict $ JSON.encodePretty value)
+
+sourceCodeExport :: Text -> SourceCode -> Text
+sourceCodeExport name (SourceCode value) = multilineString name value
+
 psModule :: Text -> Text -> Text
 psModule name body = "module " <> name <> " where" <> body
 
+------------------------------------------------------------
 writeUsecases :: FilePath -> IO ()
 writeUsecases outputDir = do
     let usecases =
-            multilineString "vesting" vesting <> multilineString "game" game <>
-            multilineString "crowdfunding" crowdfunding <>
-            multilineString "errorHandling" errorHandling <>
-            multilineString "starter" starter
+            sourceCodeExport "vesting" vesting <> sourceCodeExport "game" game <>
+            sourceCodeExport "crowdFunding" crowdFunding <>
+            sourceCodeExport "errorHandling" errorHandling <>
+            sourceCodeExport "starter" starter <>
+            jsonExport "contractDemos" contractDemos
         usecasesModule = psModule "Playground.Usecases" usecases
     BS.writeFile
         (outputDir </> "Playground" </> "Usecases.purs")
         (T.encodeUtf8 usecasesModule)
-    putStrLn outputDir
 
+------------------------------------------------------------
+writeTestData :: FilePath -> IO ()
+writeTestData outputDir = do
+    let ContractDemo { contractDemoContext
+                     , contractDemoSimulations
+                     , contractDemoEditorContents
+                     } = head contractDemos
+    BSL.writeFile
+        (outputDir </> "compilation_response.json")
+        (JSON.encodePretty contractDemoContext)
+    void $
+        itraverse
+            (\index ->
+                 writeSimulation
+                     (outputDir </> "evaluation_response" <> show index <>
+                      ".json")
+                     contractDemoEditorContents)
+            contractDemoSimulations
```

Comment:
Wait, so this executable now generates the test output too? I find that somewhat confusing...

Reply:
Maybe I'm just being grumpy, but it would be nice if `PSGenerator` just generated `PS` ;)

Reply:
Well, it generates everything you need for the PureScript codebase. I _could_ split it into two executables, but then you'd always want to run both at the same time, one after the other. That doesn't feel like an improvement.

Perhaps a name-change would make you feel better about it? 🤔

Reply:
It's fine, it just feels vaguely untidy.